### DISCUSSION TITLE
IPhone Execution is Slower Compare to IPAD Execution Using Appium 2.0

### DISCUSSION THREAD

**Discussion by Manikantan1991** [appium/appium#19246] [Q&A]
The Framework is keyword driven, and utilising an Excel sheet and keywords, the same test script is created and executed on both an iPad and an iPhone. The way the sheets handle swipe and scroll for both devices is the sole distinction between the different sheets we utilise for the iPad and iPhone test cases.

But in theory, an IPAD’s execution time would be quicker than an iPhone’s. Note: The network bandwidth is the same and the same programme is running on both devices.

Iphone Specifications: Iphone 13 Pro with iOS 16.6.1 and 6GB of RAM. Appium 2.0

iPad Specifications: iPAd Air (4th generation) iOS 16.3.1, 4GB of RAM, and 2.0 Appium Has this issue happened to anyone else?

Tried by running the Testcases bulk and even set wise but still exeution time is more compare to IPAD execution .

**Comment by jlipps** [comment]
No real idea but what I would do to debug is run the appium server with `--log-timestamp` turned on and examine the flow of each device to see where discrepancies in time show up.
